Bug fix

boygear opened the pull request.

Commit:
changed date format

Commit:
fixed date format bug

Commit:
bug with resize

Commit:
bug fixes

axandar commented on src/main/java/pl/nowekolory/objectifyxlsx/cell/CellCreator.java:
```diff
@@ -80,9 +78,13 @@ private void addCell(Row row, Integer value, Integer cellIndex, CellStyle cellSt
-        var cell = row.createCell(cellIndex);
-        cell.setCellValue(value);
-        cell.setCellStyle(cellStyle);
+        if(value == 0.0){
```
Czemu ten fix jest tylko dla typu Double? Float też nie powinien tego mieć? Inne wartości liczbowe nie potrzebują?

boygear replied:
Paweł chciał, żeby ujednolicić to, że jest jest 0.0 a w niektórych nic to ma być wszędzie nic. Osobiście uważam, że to nie jest dobre wsadzanie wszędzie pustych miejsc zamiast 0. Spytam jeszcze Pawła o to. Float zmienię bo to podobna sytuacja, ale swoją drogą nigdzie chyba nie używam tego typu.

axandar replied:
Dodaj to jako task i psytaj Pawła

axandar commented on src/main/java/pl/nowekolory/objectifyxlsx/cell/CellStyleCreator.java:
```diff
@@ -3,11 +3,14 @@
 
-    private static final String defaultDateFormat = "dd.MM.yyyy";
-    private static final String defaultDateTimeFormat = "dd.MM.yyyy h:mm";
+    private static final String defaultDateFormat = "M.d.yy";
```
Zapisz dokumentację w kodzie, że jest to wewnętrzny wzór excela i nie można tego zmieniać.

boygear replied:
Sprawdziłem to i np: d.M.yy też działa, więc zmieniłem na taki bo jest czytelniejszy. Mam wrażenie, że excel jest w stanie złapać praktycznie dowolny format zgodny ze standardami. Zostawię tam informację, że format powinien być zgodny ze standardem Excela

axandar commented on src/main/java/pl/nowekolory/objectifyxlsx/cell/CellStyleCreator.java:
```diff
@@ -36,7 +39,7 @@ public static CellStyle createDateTimeCellStyle(Workbook workbook, String format
     public static CellStyle createDateCellStyle(Workbook workbook, String format){
         var dateCellStyle = workbook.createCellStyle();
-        var stringFormat = getFormat(format, defaultDateFormat);
+        var stringFormat = getFormat(format, DateFormatConverter.convert(Locale.getDefault(), "dd.MM.yyyy"));
```
Nie używajmy Locale.getDefault - taki kod jest zależny od systemu. Narzucajmy jakiś standardowy dla nas lub dla typu danych.

boygear replied:
Zmienione na pl

axandar replied:
Nie możemy ot tak zmienić na PL. Działa to na serwerze? 
Przy domyslnym Locale nigdy nie wiemy co będzie na serwerze wybrane jako domyślne. Przy narzuceniu "PL" nie mamy pewności że Locale PL jest zainstalowane.
en_US powinno być wszędzie wspierane, możemy takiego użyć?

Commit:
Code improvement

axandar committed:
Reorganize dependencies in pom.xml

axandar committed:
Code reformatting

Commit:
locale changed

axandar committed:
Refactor CellStyleCreator for more efficient locale usage

Locale object 'usLocale' creation was moved up in the class hierarchy for improved code readability and efficiency. In the previous implementation, the 'usLocale' object was created every time the 'createDateCellStyle' function was called, leading to wastage of computation resources. Now, one instance is created once and reused throughout the class, improving overall performance.

axandar committed:
Update pom.xml version

The version in the pom.xml file was updated from 1.1.4-SNAPSHOT to 1.1.4. This change indicates the completion of development for this version and preparation for a stable release.
